python/semantic_kernel/data/vector_store_record_collection.py

```diff
@@ -0,0 +1,523 @@
+
+        try:
+            data = self.serialize(record)
+        except Exception as exc:
+            raise MemoryConnectorException(f"Error serializing record: {exc}") from exc
+
+        try:
+            results = await self._inner_upsert(data if isinstance(data, Sequence) else [data], **kwargs)
+        except Exception as exc:
+            raise MemoryConnectorException(f"Error upserting record: {exc}") from exc
+
+        if self._container_mode:
+            return results
+        return results[0]
+
+    async def upsert_batch(
+        self,
+        records: OneOrMany[TModel],
+        embedding_generation_function: Callable[
+            [VectorStoreRecordDefinition, OneOrMany[TModel]], Awaitable[OneOrMany[TModel]]
+        ]
+        | None = None,
+        **kwargs: Any,
+    ) -> Sequence[TKey]:
+        """Upsert a batch of records.
+
+        Args:
+            records (Sequence[TModel] | TModel): The records to upsert, can be a list of records, or a single container.
+            embedding_generation_function (Callable): Supply this function to generate embeddings.
+                This will be called with the data model definition and the records,
+                should return the records with vectors.
+                This can be supplied by using the add_vector_to_records method from the VectorStoreRecordUtils.
+            **kwargs (Any): Additional arguments.
+
+        Returns:
+            Sequence[TKey]: The keys of the upserted records, this is always a list,
+            corresponds to the input or the items in the container.
+        """
+        if embedding_generation_function:
+            records = await embedding_generation_function(self.data_model_definition, records)
+
+        try:
+            data = self.serialize(records)
+        except Exception as exc:
+            raise MemoryConnectorException(f"Error serializing records: {exc}") from exc
+
+        try:
+            return await self._inner_upsert(data, **kwargs)  # type: ignore
+        except Exception as exc:
+            raise MemoryConnectorException(f"Error upserting records: {exc}") from exc
+
+    async def get(self, key: TKey, **kwargs: Any) -> TModel | None:
+        """Get a record.
+
+        Args:
+            key (TKey): The key.
+            **kwargs (Any): Additional arguments.
+
+        Returns:
+            TModel: The record.
+        """
+        try:
+            records = await self._inner_get([key])
+        except Exception as exc:
+            raise MemoryConnectorException(f"Error getting record: {exc}") from exc
+
+        if not records:
+            return None
+
+        try:
+            model_records = self.deserialize(records[0], keys=[key], **kwargs)
+        except Exception as exc:
+            raise MemoryConnectorException(f"Error deserializing record: {exc}") from exc
+
+        if isinstance(model_records, Sequence):
+            return model_records[0]
+        return model_records
```

Comment:
Question: why is it necessary to check if the deserialized `records[0]` is a `Sequence`? If we are only returning the first element of the sequence, it seems like we are only returning part of the first record.

Reply:
added a comment about the logic, has to do with container mode and what deserialize returns.